compiler/src/main/scala/dx/compiler/ApplicationCompiler.scala

```diff
@@ -440,24 +440,37 @@ case class ApplicationCompiler(typeAliases: Map[String, Type],
       case ExecutableKindWorkflowOutputReorg =>
         // The reorg applet requires higher permissions to organize the output directory.
         Some(DxAccess.empty.copy(project = Some(DxAccessLevel.Contribute)))
+      case ExecutableKindWfInputs(blockPath) =>
+        Some(
+            DxAccess.empty.copy(project = Some(DxAccessLevel.Upload),
+                                allProjects = Some(DxAccessLevel.View))
+        )
+      case ExecutableKindWfOutputs(blockPath) =>
+        Some(DxAccess.empty.copy(allProjects = Some(DxAccessLevel.Upload)))
       case _ =>
         // Scatters need network access, because they spawn subjobs that (may) use dx-docker.
         // We end up allowing all applets to use the network
-        Some(DxAccess.empty.copy(network = Some(Vector("*"))))
+        Some(DxAccess.empty)
     }
     // using manifests requires at least UPLOAD access
     val manifestAccess = if (useManifests) {
       Some(DxAccess.empty.copy(project = Some(DxAccessLevel.Upload)))
     } else {
       None
     }
+    val asDeveloperAccess = (applet.kind match {
+      case ExecutableKindNative(_, _, _, _, _) =>
+        Some(DxAccess.empty)
+      case _ => Some(DxAccess.empty.copy(developer = Some(true)))
```

Comment:
> If the user launches app1 from the UI or via 'dx run', at that time they are full scoped. The job gets exactly "full scope" ⋂ "dxapp1.json" (where '⋂' means intersection). Then when that job launches app2, the second job gets "full scope" ⋂ "dxapp1.json" ⋂ "dxapp2.json" , etc
And since "full scope" ⋂ "X" is equal to "X", in practice it's the first app that sets the tone
so I added `developer` scope to all non-native executables. But can be removed from this PR.
@commandlinegirl

Reply:
Also from George:
> It can wrap any DNAnexus native app
And that app may have all sorts of permissions requested in its dxapp.json
The wrapper app(let) MUST have at least that much of permissions in its own dxapp.json, otherwise the system cannot "elevate" permissions from one job to the next
So if the wrapper has lesser permissions, the app that is launched will inherit those stricter permissions

So I wonder if keeping the scope to be `developer` would be something useful...